pythia/models/butd.py

```diff
@@ -0,0 +1,294 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+import gc
+import torch
+from pythia.common.registry import registry
+from pythia.modules.encoders import ImageEncoder
+from pythia.modules.decoders import CaptioningDecoder
+from torch import nn
+
+from .base_model import BaseModel
+
+
+@registry.register_model("butd")
+class BUTD(BaseModel):
+    def __init__(self, config):
+        super().__init__(config)
+        self.config = config
+        self._global_config = registry.get("config")
+        self._datasets = self._global_config.datasets.split(",")
+
+    def build(self):
+        self._build_word_embedding()
+        self._init_feature_encoders("image")
+        self._init_decoder()
```

Comment:
I will try to modularize this and move the `TopDownAttention` part outside the decoder and make the decoder a configurable from configs. I will add these changes after I fix the other ones as this will require retraining the models and checking if we maintain the accuracy after the refactor.